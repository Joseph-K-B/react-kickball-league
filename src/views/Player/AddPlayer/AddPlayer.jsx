import { useState } from "react"
import { useHistory } from "react-router-dom"
import { createPlayer } from "../../../services/players";
import PlayerForm from "../../../components/PlayerForm";

function AddPlayer() {
    const [name, setName] = useState('');
    const [position, setPosition]= useState('');
    const [teamId, setTeamId] = useState(null);
    const history = useHistory();

    
    

    const handleAddPlayerSubmit = async (e) => {
        e.preventDefault();
        console.log('TEAM ID', teamId);
        const res = await createPlayer({name, position, teamId})
        history.push(`/players/${res[0].id}`)
    }

    // console.log(teamId);

    return(
        <>
        <fieldset>
            <legend>Add Player</legend>
            <PlayerForm 
                handleSubmit={handleAddPlayerSubmit}
                name={name}
                position={position}
                teamId={teamId}
                setName={setName}
                setPosition={setPosition}
                setTeamId={setTeamId}
            />
        </fieldset>
        </>
    )
}

export default AddPlayer;