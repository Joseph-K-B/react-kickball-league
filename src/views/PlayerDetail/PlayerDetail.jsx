import { useEffect, useState } from "react"
import { Link, useParams } from "react-router-dom"
import { getPlayerById } from "../../services/players"

function PlayerDetail() {
    const{ id } = useParams()
    const[player, setPlayer] = useState(null)

    useEffect(() => {
        getPlayerById(id).then((res) => setPlayer(res))
    }, [id])

    if(!player) return <h2>Loading player info...</h2>

    return(
        <>
        <h1>{player.name}</h1>
        <Link to={`/teams/${player.teams.id}`}>
        <h3>{player.teams.name}</h3>
        </Link>
        <h3>{player.teams.city}, {player.teams.state}</h3>        
        <h3>{player.position}</h3>
        </>
    )
}
export default PlayerDetail