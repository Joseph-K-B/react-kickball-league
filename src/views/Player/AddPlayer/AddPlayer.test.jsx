import { render, screen } from '@testing-library/react';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import {Router, Route } from 'react-router-dom';
import { createMemoryHistory } from "history";
import userEvent from '@testing-library/user-event';
import AddPlayer from './AddPlayer';
import PlayerDetail from '../PlayerDetail/PlayerDetail';
import { act } from 'react-dom/test-utils';

const mockPlayer = {
    id: 12,
    created_at: '2021-12-08T16:09:33.408898+00:00',
    name: 'player name',
    position: 'benchwarmer',
    teamId: 1
}

const mockTeam = {
    id: 1,
    created_at: '2021-12-08T16:09:33.408898+00:00',
    name: 'name',
    city: 'city',
    state: 'state',
    players: []
}

const mockTeam2 = {
        id: 2,
        created_at: '2021-12-08T16:09:33.408898+00:00',
        name: 'team name two',
        city: 'city two',
        state: 'state',
        players: []
    }

const server = setupServer(
    rest.get('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/teams', 
    (req, res, ctx) => {
        return res(ctx.json(mockTeam));
    }),
    rest.post('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/teams', 
    (req, res, ctx) => {
        return res (
            ctx.json([mockTeam]));
    }),
    rest.get('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/teams', 
    (req, res, ctx) => {
        return res(ctx.json(mockTeam2));
    }),
    rest.post('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/teams', 
    (req, res, ctx) => {
        return res (
            ctx.json([mockTeam2]));
    }),
    rest.get('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/players', 
    (req, res, ctx) => {
        return res(ctx.json(mockPlayer));
    }),
    rest.post('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/players', 
    (req, res, ctx) => {
        return res (
            ctx.json([mockPlayer]));
    })
);

beforeAll(() => {
    server.listen();
});

afterAll(() => {
    server.close();
});

it('renders form to allow admin to add player and redirects to their details', async () => {
    const history = createMemoryHistory();
    history.push('/players/new');

    const {container} = render (
        <Router history={history}>
            <Route exact path = '/players/new'>
                <AddPlayer />
            </Route>
            <Route exact path = '/players/:id' component={PlayerDetail} />
        </Router>
    );

    screen.getByText('Add Player');

    const nameField = screen.getByLabelText(/Name:/);
    const positionField = screen.getByLabelText(/Position:/);
    const submitBtn = screen.getByRole('button', {name: 'edit player'});
    // const teamField = await screen.findByLabelText(/Team:/);
    const optionChoice = await screen.findByRole('combobox');

    act(async () => {
        userEvent.type(nameField, 'New Team');
        userEvent.type(positionField, 'cheerleader');
        userEvent.selectOptions(optionChoice, [mockTeam.id]);
        userEvent.click(submitBtn);
    })

    // expect(screen.getByRole('option', {name: mockTeam.name}).selected).toBe(true)
    expect(container).toMatchSnapshot();
    // await screen.findByText('player name');
});