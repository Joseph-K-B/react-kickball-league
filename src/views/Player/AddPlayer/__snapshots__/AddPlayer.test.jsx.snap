// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form to allow admin to add player and redirects to their details 1`] = `
<div>
  <fieldset>
    <legend>
      Add Player
    </legend>
    <form>
      <label
        for="name"
      >
        Name:
      </label>
      <input
        id="name"
        type="text"
        value="m"
      />
      <label
        for="position"
      >
        Position:
      </label>
      <input
        id="position"
        type="text"
        value="r"
      />
      <label
        for="teams"
      >
        Team:
      </label>
      <select
        aria-label="teams"
        id="teams"
        name="select team"
      />
      <button
        aria-label="edit player"
        name="edit player"
        type="submit"
      >
        Submit
      </button>
    </form>
  </fieldset>
  <h2>
    Loading player info...
  </h2>
</div>
`;
